lang/java/avro/src/test/java/org/apache/avro/TestSchemaCompatibilityMissingUnionBranch.java

```diff
@@ -101,11 +107,16 @@
   @Parameter(1)
   public Schema writer;
   @Parameter(2)
-  public String details;
+  public List<String> details;
+  @Parameter(3)
+  public List<String> location;
 
   @Test
   public void testMissingUnionBranch() throws Exception {
-    validateIncompatibleSchemas(reader, writer, SchemaIncompatibilityType.MISSING_UNION_BRANCH,
-        details);
+    List<SchemaIncompatibilityType> types = new ArrayList<SchemaCompatibility.SchemaIncompatibilityType>(details.size());
```

Comment:
nit: I think we create a list with N `SchemaIncompatibilityType.MISSING_UNION_BRANCH` items in a single line using `java.util.Collections#nCopies` The return value is immutable, but in this case it is not an issue.